### DISCUSSION TITLE
Using OpenTelemetry Collector to send traces collected out of process

### DISCUSSION THREAD

**Discussion by MacNale** [open-telemetry/opentelemetry-collector#2447] [Q&A] [answered]
Team,

I am collecting traces in a kafka topic (instrumented out of process without using auto instrumentation agents). The traces are not in standard OLTP format. I wish to read the trace and map it to OLTP format. Then using collector ingest into any supported backend.

Application --produces traces on a Kafka topic-> I WANT TO CONVERT TO OLTP format -> OTLP Collector-->Backend

I am not sure how should I get started. I generated code from proto files. Now how can I create each trace/span in OLTP format, so that I can see in Jeager.

Thanks,
Mac

**Comment by bogdandrutu** [comment, marked as answer]
Someone is already working at https://github.com/open-telemetry/opentelemetry-collector/pull/2467, and this will have a similar mechanism to plugin custom formats as we have for the [kafka exporter](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/kafkaexporter/factory.go#L44).

So right now it is not available but soon we will have that.

**Reply by MacNale** [reply]
The PR ("kafkametrics receiver ) talks about receiving metrics. I am looking for traces.

**Reply by bogdandrutu** [reply]
That can be extended to support traces as well.

**Reply by bogdandrutu** [reply]
I missed link the right receiver. Can you check this https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/kafkareceiver
